### PR TITLE
Add automatic SDK updates

### PR DESCRIPTION
**Pull request by wtgodbe**
Add a workflow to automatically update the SDK on Monday. Will update the build-ops docs once this is in.

CC @blowdart we're pinning to the `v3.1.0` tag per your request

### CODE REVIEW

**Comment by wtgodbe** [comment]
I'll test out the workflow once it's in main.

**Comment by blowdart** [comment]
Please pin to the commit ID not the version

See https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

File: .github/workflows/update-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  # Manual trigger to update the .NET SDK on-demand.
+  workflow_dispatch:
+
+jobs:
+  update-dotnet-sdk:
+    name: Update .NET SDK
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: martincostello/update-dotnet-sdk@v3.1.0
```
Comment (blowdart) [inline .github/workflows/update-sdk.yml:18]: - uses: martincostello/update-dotnet-sdk@8765302da88e1a78c845125626f3726190908593 # v3.1.0

Commit is from https://github.com/martincostello/update-dotnet-sdk/commit/8765302da88e1a78c845125626f3726190908593

**Review by blowdart — CHANGES_REQUESTED** [review]
Pinning to commit ID not version

**Review by blowdart — APPROVED** [review]

**Pull request merged**